### PR TITLE
Navbar implementation

### PR DESCRIPTION
**Pull request by GuillaumeRx**

### CODE REVIEW

File: src/theme/theme.ts [hunk 1]
```diff
@@ -111,20 +125,41 @@ export const textVariants = variant({
...
-  verticalCenter: {
+  center: {
     height: '100%',
-    display: 'flex',
     justifyContent: 'center',
     alignItems: 'center'
   }
 };
 
+const FLEX_VARIANTS = {
```
Comment (vilsbole) [inline src/theme/theme.ts:140]: 😍

File: src/features/Layout/Navigation/constants.ts [hunk 2]
```diff
@@ -0,0 +1,262 @@
+import { EXT_URLS } from '@config';
+import { IAppRoutes } from '@routing/routes';
+import { translateRaw } from '@translations';
+import { INavTray, IRouteLink } from '@types';
+
+//const APP_ROUTES_OBJECT = getAppRoutesObject(featureFlags);
```
Comment (vilsbole) [inline src/features/Layout/Navigation/constants.ts:6]: deadcode

File: src/features/Layout/Navigation/DesktopNav.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import { COLORS } from '@theme';
+
+import { Navbar, NavLink, NavTray, SupportUsTray, TrayItem } from './components';
+import { desktopLinks, settingsLinks, supportUsTray, toolsTray } from './constants';
+
+const DesktopNav = () => {
+  const { pathname } = useLocation();
+  const { featureFlags } = useFeatureFlags();
+
+  const APP_ROUTES = getAppRoutesObject(featureFlags);
```
Comment (vilsbole) [inline src/features/Layout/Navigation/DesktopNav.tsx:17]: How about passing the active routes as a prop ?

File: src/features/Layout/Navigation/MobileNav.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import { getAppRoutesObject } from '@routing';
+import { useFeatureFlags } from '@services';
+
+import { Navbar, NavLink, NavTray, TrayItem } from './components';
+import { mobileLinks } from './constants';
+
+const MobileNav = () => {
+  const { pathname } = useLocation();
+  const { featureFlags } = useFeatureFlags();
+  const APP_ROUTES = getAppRoutesObject(featureFlags);
```
Comment (vilsbole) [inline src/features/Layout/Navigation/MobileNav.tsx:14]: How about passing active route as prop ?

File: src/features/Layout/Navigation/components/NavTray.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+import { Tray } from './Tray';
+
+const SBox = styled(Box)`
+  cursor: pointer;
+`;
+
+const SText = styled(Text)`
+  text-transform: uppercase;
+`;
```
Comment (vilsbole) [inline src/features/Layout/Navigation/components/NavTray.tsx:17]: Can we use the `sx` prop for simple changes like these?
I avoid the extra declaration

File: src/features/Layout/Navigation/components/Navbar.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    <Box
+      position="fixed"
+      zIndex={999}
+      backgroundColor="BLUE_DARK_SLATE"
+      width={{ _: '100vw', sm: '64px' }}
+      height={{ _: '57px', sm: '100vh' }}
+      bottom={{ _: 0, sm: 'auto' }}
+      top={{ sm: 0 }}
+    >
+      <Box variant="rowAlign" flexDirection={{ sm: 'column' }}>
```
Comment (vilsbole) [inline src/features/Layout/Navigation/components/Navbar.tsx:16]: 😍

File: src/features/Layout/Navigation/components/SupportUsTray.tsx [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+import { ExternalLink } from './ExternalLink';
+
+const SText = styled(Text)`
+  text-transform: uppercase;
+`;
+
+export const SupportUsTray = ({ items }: { items: TTrayItem[] }) => {
+  const [displayingMessage, setDisplayMassage] = useState(false);
+  const timeoutId = useRef<ReturnType<typeof setTimeout> | null>(null);
+  const trackDoncation = useAnalytics({
```
Comment (vilsbole) [inline src/features/Layout/Navigation/components/SupportUsTray.tsx:23]: typo `Donation`

File: src/features/Layout/Navigation/components/SupportUsTray.tsx [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+  }, [timeoutId]);
+
+  useEffect(() => {
+    return () => {
+      clearTimeoutId();
+    };
+  }, [clearTimeoutId]);
+
+  const displayMessage = useCallback(() => {
+    setDisplayMassage(true);
```
Comment (vilsbole) [inline src/features/Layout/Navigation/components/SupportUsTray.tsx:41]: typo `Message`

File: src/features/Layout/Navigation/components/SupportUsTray.tsx [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+    if (timeoutId.current) {
+      clearTimeout(timeoutId.current);
+      timeoutId.current = null;
+    }
+  }, [timeoutId]);
+
+  useEffect(() => {
+    return () => {
+      clearTimeoutId();
+    };
```
Comment (vilsbole) [inline src/features/Layout/Navigation/components/SupportUsTray.tsx:37]: Can be written `return clearTimeoutId`

File: src/features/Layout/Navigation/components/SupportUsTray.tsx [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+
+import { ExternalLink } from './ExternalLink';
+
+const SText = styled(Text)`
+  text-transform: uppercase;
+`;
+
+export const SupportUsTray = ({ items }: { items: TTrayItem[] }) => {
+  const [displayingMessage, setDisplayMassage] = useState(false);
+  const timeoutId = useRef<ReturnType<typeof setTimeout> | null>(null);
```
Comment (vilsbole) [inline src/features/Layout/Navigation/components/SupportUsTray.tsx:22]: Prefer `useTimeout` https://github.com/streamich/react-use/blob/master/docs/useTimeout.md
We used to copy selected hooks into `@vendor`.

File: src/features/Layout/Navigation/components/TrayItem.tsx [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+import styled from 'styled-components';
+
+import { Box, Icon, Text } from '@components';
+import { COLORS } from '@theme';
+import { IRouteLink } from '@types';
+
+const SLink = styled(Link)`
+  width: 100%;
+  margin: 8px 0;
+`;
```
Comment (vilsbole) [inline src/features/Layout/Navigation/components/TrayItem.tsx:13]: Prefer props

File: src/features/Layout/Navigation/components/SupportUsTray.tsx [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+import { ANALYTICS_CATEGORIES } from '@services';
+import { SPACING } from '@theme';
+import { translateRaw } from '@translations';
+import { TTrayItem } from '@types';
+
+import { ExternalLink } from './ExternalLink';
+
+const SText = styled(Text)`
+  text-transform: uppercase;
+`;
```
Comment (vilsbole) [inline src/features/Layout/Navigation/components/SupportUsTray.tsx:18]: Prefer `sx`

File: src/types/navigation.ts [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+
+export interface IExternalLink {
+  type: 'external';
+  title: string;
+  link: TURL;
+  analyticsEvent: string;
+  icon: TIcon;
+}
+
+export type TTrayItem = IRouteLink | IExternalLink;
```
Comment (vilsbole) [inline src/types/navigation.ts:21]: I'm wondering if we can save ourself the typings.
Can TS infer these types ? Have you considered creating a common interface for both ?

**Review by vilsbole — CHANGES_REQUESTED** [review]

File: src/features/Layout/Navigation/components/NavTray.tsx (hunk 5 above)
Comment (GuillaumeRx) [reply src/features/Layout/Navigation/components/NavTray.tsx:17]: We can't use `sx` in `styled-system` it's something from `rebass` :(

Comment (GuillaumeRx) [reply src/features/Layout/Navigation/components/NavTray.tsx:17]: I'll use `style` instead

File: src/types/navigation.ts (hunk 13 above)
Comment (GuillaumeRx) [reply src/types/navigation.ts:21]: TS can't infer these types and I need those on Component props. I've considered using a common interface for both but it adds a lot of conditional on components and sometimes it's not even usable.

Using discriminated unions was the cleanest way to define those

**Comment by vilsbole** [comment]
Magical! If you can just deactivate the new nav flag then we can merge.

**Review by vilsbole — APPROVED** [review]

**Pull request merged**